agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -416,6 +416,23 @@ void ap_manager_thread::after_select(bool timeout)
                 now + std::chrono::seconds(HEARTBEAT_NOTIFICATION_DELAY_SEC);
         }
     }
+
+    // check if any client disallow period has expired and allow it.
+    if (!m_disallowed_clients.empty()) {
+        const size_t list_length = m_disallowed_clients.size();
+        for (size_t i = 0; i != list_length; ++i) {
+            if (std::chrono::steady_clock::now() > m_disallowed_clients[i].timeout) {
+
+                LOG(DEBUG) << "CLIENT_ALLOW: mac = " << m_disallowed_clients[i].mac
+                           << ", bssid = " << m_disallowed_clients[i].bssid;
+                // send allow client msg to bssid
+                ap_wlan_hal->sta_allow(network_utils::mac_to_string(m_disallowed_clients[i].mac),
+                                       network_utils::mac_to_string(m_disallowed_clients[i].bssid));
+                // erase client from disallowed list
+                m_disallowed_clients.erase(m_disallowed_clients.begin() + i);
+            }
+        }
+    }
```

Comment:
I don't understand why this has to be so complicated...
```suggestion
    for (auto it = m_disallowed_clients.begin; it != m_disallowed_clients.end(); ) {
        if (std::chrono::steady_clock::now() > it->timeout) {
            LOG(DEBUG) << "CLIENT_ALLOW: mac = " << m_disallowed_clients[i].mac
                           << ", bssid = " << m_disallowed_clients[i].bssid;
            // allow client on bssid
            ap_wlan_hal->sta_allow(network_utils::mac_to_string(it->mac),
                                   network_utils::mac_to_string(it->bssid));
            // erase client from disallowed list
            m_disallowed_clients.erase(m_disallowed_clients.begin() + i);
        } else {
            it++;
        }
    }
```